presets/inference/text-generation/inference_api_vllm.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import logging
+import os
+import asyncio
+
+import uvloop
+from pydantic import BaseModel, Field
+from fastapi import Request, HTTPException
+
+from vllm.utils import FlexibleArgumentParser
+import vllm.entrypoints.openai.api_server as api_server
+
+# Initialize logger
+logger = logging.getLogger(__name__)
+debug_mode = os.environ.get('DEBUG_MODE', 'false').lower() == 'true'
+logging.basicConfig(level=logging.DEBUG if debug_mode else logging.INFO)
+
+class HealthStatus(BaseModel):
+    status: str = Field(..., example="Healthy")
+
+@api_server.router.get("/healthz")
+async def health_check(raw_request: Request) -> HealthStatus:
+    """Health check."""
+    try:
+        await asyncio.wait_for(api_server.engine_client(raw_request).check_health(), timeout=1.0)
+        return HealthStatus(status="Healthy")
+    except asyncio.TimeoutError:
+        raise HTTPException(status_code=500, detail="Model not initialized")
```

Comment:
does vllm already come with health endpoint?

Reply:
Yes, but I don't want to expose this difference to controller part.